src/thor/costmatrix.cc

```diff
@@ -45,7 +45,9 @@ inline const valhalla::PathEdge& find_correlated_edge(const valhalla::Location&
 namespace valhalla {
 namespace thor {
 
-class CostMatrix::ReachedMap : public robin_hood::unordered_map<uint64_t, std::vector<uint32_t>> {};
+// segmented_map has much lower memory requirement; no re-allocation is happening
+class CostMatrix::ReachedMap
+    : public ankerl::unordered_dense::segmented_map<uint64_t, std::vector<uint32_t>> {};
```

Comment:
seems much nicer for these HUGE maps which probably re-allocate constantly: https://github.com/martinus/unordered_dense/tree/main?tab=readme-ov-file#4-segmented_map-and-segmented_set. pity the "benchmark" didn't include robin-hood-hashing